components/inputs/input-text.js

```diff
@@ -84,17 +162,29 @@ class InputText extends RtlMixin(LitElement) {
 				?readonly="${this.readonly}"
 				size="${ifDefined(this.size)}"
 				step="${ifDefined(this.step)}"
+				style="${styleMap(styles)}"
 				tabindex="${ifDefined(this.tabindex)}"
 				type="${this._getType()}"
 				.value="${this.value}">
+				<slot name="left"></slot>
+				<slot name="right"></slot>
 		`;
 		if (this.label && !this.labelHidden) {
-			return html`
+			input = html`
 				<label>
 					<span class="d2l-input-label">${this.label}</span>
 					${input}
 				</label>`;
 		}
+
+		if (this._hasSlot) {
```

Comment:
To simplify the rendering and reduce the number of different permutations, does it hurt anything having the container even if there aren't any slots?

Reply:
It's probably fine. I'll run the visual-diff tests to confirm.